File: livingrimoire start here/LivinGrimoire java/src/LivinGrimoire/Fusion.java
package LivinGrimoire;

public class Fusion {
    private String emot = "";
    private String result = "";
    private Cerabellum[] ceraArr = new Cerabellum[5];

    public Fusion() {
        for (int i = 0; i < 5; i++) {
            ceraArr[i] = new Cerabellum();
        }
    }

    public String getEmot() {
        return emot;
    }
    public void loadAlgs(Neuron neuron){
        for (int i = 1; i < 6; i++) {
            if (!ceraArr[i-1].isActive()){
                Algorithm temp = neuron.getAlg(i);
                if (temp != null){
                    ceraArr[i-1].setAlgorithm(temp);
                }
            }
        }
    }
    public String runAlgs(String ear, String skin, String eye){
        result = "";
        for (int i = 0; i < 5; i++) {
            if (!ceraArr[i].isActive()){
                continue;
            }
            result = ceraArr[i].act(ear,skin,eye);
            ceraArr[i].advanceInAlg();
            emot = ceraArr[i].getEmot();
            ceraArr[i].deActivation(); // deactivation if Mutatable.algkillswitch = true
            return result;
        }
        emot = "";
        return result;
    }
}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/livinGrimoire/Fusion.kt
package livinGrimoire

class Fusion {
    var emot = ""
        private set
    private var result = ""
    private val ceraArr = arrayOfNulls<Cerabellum>(5)

    init {
        for (i in 0..4) {
            ceraArr[i] = Cerabellum()
        }
    }

    fun loadAlgs(neuron: Neuron) {
        for (i in 1..5) {
            if (!ceraArr[i - 1]!!.isActive) {
                val temp = neuron.getAlg(i)
                if (temp != null) {
                    ceraArr[i - 1]!!.setAlgorithm(temp)
                }
            }
        }
    }

    fun runAlgs(ear: String?, skin: String?, eye: String?): String {
        result = ""
        for (i in 0..4) {
            if (!ceraArr[i]!!.isActive) {
                continue
            }
            result = ceraArr[i]!!.act(ear!!, skin!!, eye!!)
            ceraArr[i]!!.advanceInAlg()
            emot = ceraArr[i]!!.emot
            ceraArr[i]!!.deActivation() // deactivation if Mutatable.algkillswitch = true
            return result
        }
        emot = ""
        return result
    }
}